osu.Game/Beatmaps/IBeatmapResourceProvider.cs

```diff
@@ -16,6 +16,11 @@ public interface IBeatmapResourceProvider : IStorageResourceProvider
         /// </summary>
         TextureStore LargeTextureStore { get; }
 
+        /// <summary>
+        /// Retrieve a global large texture store, used specifically for retrieving cropped beatmap panel backgrounds.
+        /// </summary>
+        TextureStore BeatmapPanelTextureStore { get; }
```

Comment:
I kinda hate to see this sort of thing on a public interface. Do you think we could make this `internal`?

Rationale is that it is our optimisation path using this and I don't want anyone else latching onto this for some reason (unless that reason is good), while we're specifically cropping backgrounds to specific sizes for the uses of the carousel.

Reply:
Hmm, it's only used in internal usages, so we could make the whole interface `internal` instead?

Reply:
Works for me.